utils/post_processing/restructure_borealis.py

```diff
@@ -0,0 +1,329 @@
+# Copyright 2019 SuperDARN Canada, University of Saskatchewan
+# Author: Liam Graham
+
+#
+# restructure_borealis.py
+# 2019-07-30
+# Command line tool for strucuturing Borealis files
+# for bfiq, antennas iq, and raw acf data into a smaller,
+# faster, and more usable format
+
+import deepdish as dd
+import numpy as np
+import sys
+
+
+def find_shared(data):
+	"""
+	Finds fields in an hdf5 file that do not change
+	between records
+	"""
+	shared = list()
+	unshared = list()
+	start = True
+
+	for k in data:
+		if start:
+			data_sub = data[k]
+			start = False
+		else:
+			for f in data[k]:
+				if type(data[k][f]) is np.ndarray:
+					if np.array_equal(data[k][f], data_sub[f]):
+						if f not in shared and f not in unshared:
+							shared.append(f)
+					else:
+						unshared.append(f)
+				else:
+					if data[k][f] == data_sub[f]:
+						if f not in shared and f not in unshared:
+							shared.append(f)
+					else:
+						unshared.append(f)
+	return shared
+
+def find_max_sequences(data):
+	"""
+	Finds the maximum number of sequences between records in a data file
+	"""
+	first_key = list(data.keys())[0]
+	max_seqs = data[first_key]["num_sequences"]
+	for k in data:
+		if max_seqs < data[k]["num_sequences"]:
+			max_seqs = data[k]["num_sequences"]
+	return max_seqs
+
+
+def restructure_data(data_path):
+	"""
+	Restructure the data contained in an hdf5 file to eliminate the record format.
+	Rather, data will be contained in a large array according to data dimensions.
+	Examples: for rawacfs, this array will be of shape (num_records, num_arrays, num_sequences, num_beams, num_ranges, num_lags)
+	Fields from the original record that do not change between records will be stored as fields in one metadata record within
+	the file. Other fields will contain the data arrays and other metadata that does change record to record.
+	Args:
+		data_path:	string containing the path to the data file for restructuring
+	Returns:	If valid filetype, returns None and saves the data as a newly
+				formatted hdf5 file.
+	"""
+
+	def restructure_pre_bfiq(data_record):
```

Comment:
again I'd prefer the restructure functions to just return the new dictionary so it can be used in other purposes, and the writing function to be separate. Also, the functions pulled out of the parent function so that it can be imported in other batch scripts easily.

Reply:
Changed this.